src/components/Articles.jsx

```diff
@@ -12,11 +12,20 @@ function Articles() {
   const [articles, setArticles] = useState([]);
   const [search, setSearch] = useState("");
   const [topics, setTopics] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false)
 
   useEffect(() => {
+    setLoading(true);
+    setError(false);
     getArticles().then((res) => {
       const newList = res.filter((article) => article.title.toLowerCase().includes(search.toLowerCase()));
```

Comment:
This is not currently needed for this ticket, and the user experience for the search bar is not great currently. If you want to implement this now, you need to make sure its working well. I would personally delete this and you can add it when you get to the advanced tickets.